src/StructuredLogger/BinaryLogger/Postprocessing/ReaderErrorType.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Logging.StructuredLogger;
+
+/// <summary>
+/// Type of the error that occurred during reading.
+/// </summary>
+public enum ReaderErrorType
+{
+    /// <summary>
+    /// The encountered event is completely unknown to the reader. It cannot interpret any part of it.
+    /// </summary>
+    UnkownEventType,

```

Comment:
Unknown

Reply:
Since there are more unknowns, I'd rather be specific on what unknown is this.

Unless this is a generic unknown where you don't even know what you don't know.

It's always a good practice to have the `0` be something that is invalid, to avoid issues with uninitialized values.

Reply:
@paulomorgado - Kirill was refering to a typo, it'll still mention what unknown it is - basically 'UnkownEventType' -> 'Unk**n**ownEventType'

As for an invalid member - I'd want to avoid it here - as it imposes one extra state to cumbersomely handle on the recepient side, while it's not supposed to happen. Would the current 0 be returned by accident - it'd lead to error anyways.

Reply:
Sorry. Missed the whole thing was about a typo.

My comment was that these type of enums should always have a zero value that represents uninitialized/invalid state, because in C# the literal `0` is convertible to any enum value and it's also the default value for the enum.

For most cases (as it seems in this case), that is already the intended value.